Feat: Add pagination to user list

DevdouglasGfs opened the pull request.

DevdouglasGfs committed:
Docs: Update README.md file

DevdouglasGfs committed:
Fix: Remove unused section
Remove "Instructions" section and fix the project name

DevdouglasGfs committed:
Feat: Add back-end integration for login
Integrate with the backend for validate user login

DevdouglasGfs committed:
Feat: Add back-end integration for login
Integrate with the backend for validate user login

DevdouglasGfs committed:
Feat: Add /users route and users page
Implements also Apollo Client middleware;
Rename variables and remove unused;

DevdouglasGfs committed:
Feat: Add /users route and users page
Implements also Apollo Client middleware;
Rename variables and remove unused;

DevdouglasGfs committed:
Chore: rename login-page to login

DevdouglasGfs committed:
Chore: Create dedicated file for types and mocks

DevdouglasGfs committed:
Feat: Create useGetUsers hook
It should return a list of users

DevdouglasGfs committed:
Feat: Create the users list page
This page will be responsible for displaying the list of users

DevdouglasGfs committed:
Chore: Writting correction

DevdouglasGfs committed:
Docs: Add comments to code

DevdouglasGfs committed:
Docs: Add comments to code

DevdouglasGfs committed:
Style: Add styling to pagination controls

DevdouglasGfs committed:
Chore: Add util for make logout
Change also the checkLoginStatus login to return null or the token;

DevdouglasGfs committed:
Feat: Add basic redirection
Skip the login if the user has already logged in (it has a valid token)

DevdouglasGfs committed:
Chore: Add params for the useGetUsers hook
Is possible which the useGetUsers hook receive a param of type PageInput

DevdouglasGfs committed:
Feat: Add pagination to the list of users

DevdouglasGfs commented:
# Apresentação da tela
![Screenshot_20240626-130414~2](https://github.com/DevdouglasGfs/taqtile/assets/105404079/e77af885-0198-4510-9549-68042513fbae)
![Screenshot_20240626-130617~2](https://github.com/DevdouglasGfs/taqtile/assets/105404079/2a3f39ce-ce79-4bfd-8f02-2ce928bd906b)

DevdouglasGfs committed:
Merge branch 'feature/backend-integration' into feature/users-list

racsandoval commented on src/components/login-form/index.tsx:
```diff
@@ -1,19 +1,12 @@
-  role: string
-}
-export type UserBasicPersonalData = Required<Pick<UserDTO, 'email'> & { password: string }>;
+export type UserBasicLoginData = Required<Pick<UserDto, 'email'> & { password: string }>;
```
aqui não seria mais simples e claro usar o tipo `{ email: string; password: string }`?

DevdouglasGfs replied:
Acredito que seria sim... É que gosto de ter controle prático sob as tipagens então eu movi isso para `/src/types/user.ts` em alguma PR

racsandoval replied:
não entendi pq manter da maneira que está garante controle prático da tipagem

SamuelSsc replied:
@DevdouglasGfs

DevdouglasGfs replied:
É que tendo um arquivo único só para lidar com as tipagens relacionadas as atividades do usuário eu pensei que seria mais fácil de fazer alguma mudança se as necessidades mudassem, mas talvez não seja realmente tão simples


Então vou fazer a tipagem na própria variável em que foi usada e remover do arquivo separado

racsandoval commented on src/pages/users/index.tsx:
```diff
@@ -0,0 +1,42 @@
+
+    // Define the offset of the pagination
+    const [offset, setOffset] = useState(0);
+    const { data, loading } = useGetUsers({ offset });
```
se aqui é passado apenas o offset, como você garante que uma página tem 10 usuários? e se esse valor padrão for modificado no back?

DevdouglasGfs replied:
Não cogitei essa possibilidade, vou alterar isso!

DevdouglasGfs replied:
Já alterei

racsandoval commented on src/pages/users/index.tsx:
```diff
@@ -0,0 +1,42 @@
+                    {loading && <tr><td>Carregando...</td></tr>}
+                </tbody>
+                <div className="pagination-controls">
+                    <button disabled={offset <= 0} onClick={() => setOffset(() => offset - 10)} className="pagination-controls__prev">Anterior</button>
+                    <button onClick={() => setOffset(() => offset + 10)} className="pagination-controls__next">Próximo</button>
```
uma boa prática em controles de paginação é mostrar a página atual

DevdouglasGfs replied:
Obrigado pela informação! Vou adicionar à página

DevdouglasGfs replied:
Adicionei em um novo commit

DevdouglasGfs committed:
Chore: Offset and limit parameters explicit.
Provide a default value for the offset and limit parameters.
Provide info about what is the page that the user is currently on.

DevdouglasGfs committed:
Chore: remove offset unnecessary

DevdouglasGfs commented:
Caso eu não precisar trazer ou fazer as alterações nesta PR eu aguardo por uma nova revisão ou aprovação na PR @racsandoval @SamuelSsc @walteraandrade @marcusfreirepaulino

DevdouglasGfs committed:
Chore: Use inline typing to email and password

DevdouglasGfs committed:
Style: fix code formatation

DevdouglasGfs committed:
Merge branch 'feature/backend-integration' into feature/users-list

walteraandrade commented on src/App.tsx:
```diff
@@ -1,6 +1,8 @@
 const App: React.FC = () => {
     return (
-        <h1>HOME PAGE</h1>
+        <main>
```
Não sei se está faltando um rebase, mas o arquivo ainda continua nesse pr

DevdouglasGfs replied:
Essa página eu removi na branch `feature/user-details-page`, mas vou remover ela a partir daqui também

DevdouglasGfs replied:
Removi e coloquei apenas um parágrafo informando as 2 rotas que podem ser acessadas que tem algum conteúdo. Assim está melhor?

walteraandrade commented on src/index.tsx:
```diff
@@ -1,20 +1,37 @@
+const authLink = setContext((_, { headers }) => {
+  // get the authentication token from local storage if it exists
+  const token = localStorage.getItem('access-token');
+  // return the headers to the context so httpLink can read them
```
No geral, acho que comentários são melhores quando o código não deixa claro o que ele está fazendo, não é o caso aqui

DevdouglasGfs replied:
Certo! Removi

DevdouglasGfs committed:
Chore: Remove unrelevant comments

DevdouglasGfs committed:
Chore: Remove unused page

DevdouglasGfs committed:
Chore: Rename checkLoginStatus to getLoginToken

DevdouglasGfs committed:
Chore: Render some page instead of say which page could have content

DevdouglasGfs committed:
Style: Fix code formatting

DevdouglasGfs committed:
Style: Fix code formatting

DevdouglasGfs committed:
Merge branch 'feature/backend-integration' into feature/users-list
